solutions/src/main/java/ru/sberbank/school/task08/SerializableManager.java

```diff
@@ -16,29 +18,55 @@ public SerializableManager(@NonNull String filesDirectoryPath) {
     }
 
     @Override
-    public void initialize() {
-        throw new UnsupportedOperationException("Implement me!");
-    }
+    public void initialize() { }
 
     @Override
-    public void saveGame(String filename, Savable gameState) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+    public void saveGame(String filename, MapState<GameObject> gameState) throws SaveGameException {
+        Objects.requireNonNull(filename, "Имя файла не может быть null");
+        Objects.requireNonNull(gameState, "Состояние не может быть null");
+
+        try (OutputStream os = new FileOutputStream(filesDirectory + File.separator + filename);
+             ObjectOutputStream oos = new ObjectOutputStream(os)) {
+
+            oos.writeObject(gameState);
+
+        } catch (FileNotFoundException e) {
+            throw new SaveGameException("Отсутсвует файл", e, SaveGameException.Type.USER, gameState);
+        } catch (IOException e) {
+            throw new SaveGameException("Ошибка записи в файл", e, SaveGameException.Type.IO, gameState);
+        }
     }
 
     @Override
-    public Savable loadGame(String filename) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+    public MapState<GameObject> loadGame(String filename) throws SaveGameException {
+        Objects.requireNonNull(filename, "Имя файла не может быть null");
+        MapState<GameObject> gameState = null;
+
+        try (InputStream is = new FileInputStream(filesDirectory + File.separator + filename);
+             ObjectInputStream ois = new ObjectInputStream(is)) {
```

Comment:
&#39;try resource&#39; has incorrect indentation level 13, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1570736039>*